Userland/Libraries/LibGfx/ImageFormats/WebPLoader.cpp

```diff
@@ -297,6 +298,352 @@ static ErrorOr<VP8LHeader> decode_webp_chunk_VP8L_header(WebPLoadingContext& con
+        {4, 4}, {-4, 4}, {3, 5}, {-3, 5}, { 5, 3}, {-5, 3}, { 0, 6}, { 6, 0}, { 1, 6}, {-1, 6}, { 6, 1}, {-6, 1}, { 2, 6}, {-2, 6}, {6, 2}, {-6, 2},
+        {4, 5}, {-4, 5}, {5, 4}, {-5, 4}, { 3, 6}, {-3, 6}, { 6, 3}, {-6, 3}, { 0, 7}, { 7, 0}, { 1, 7}, {-1, 7},
+        {5, 5}, {-5, 5}, {7, 1}, {-7, 1}, { 4, 6}, {-4, 6}, { 6, 4}, {-6, 4}, { 2, 7}, {-2, 7}, { 7, 2}, {-7, 2}, { 3, 7}, {-3, 7}, {7, 3}, {-7, 3},
+        {5, 6}, {-5, 6}, {6, 5}, {-6, 5}, { 8, 0}, { 4, 7}, {-4, 7}, { 7, 4}, {-7, 4}, { 8, 1}, { 8, 2},
+        {6, 6}, {-6, 6}, {8, 3}, { 5, 7}, {-5, 7}, { 7, 5}, {-7, 5}, { 8, 4},
+        {6, 7}, {-6, 7}, {7, 6}, {-7, 6}, { 8, 5},
+        {7, 7}, {-7, 7}, {8, 6},
+        {8, 7},
+    } };
+    // clang-format on
+
+    // lz77-coded-image      =
+    //     *((argb-pixel / lz77-copy / color-cache-code) lz77-coded-image)
+    // https://developers.google.com/speed/webp/docs/webp_lossless_bitstream_specification#623_decoding_entropy-coded_image_data
+    ARGB32* pixel = context.bitmap->begin();
+    ARGB32* end = context.bitmap->end();
+
+    auto emit_pixel = [&pixel, &color_cache, color_cache_size, color_cache_code_bits](ARGB32 color) {
+        // https://developers.google.com/speed/webp/docs/webp_lossless_bitstream_specification#523_color_cache_coding
+        // "The state of the color cache is maintained by inserting every pixel, be it produced by backward referencing or as literals, into the cache in the order they appear in the stream."
+        *pixel++ = color;
+        if (color_cache_size)
+            color_cache[(0x1e35a7bd * color) >> (32 - color_cache_code_bits)] = color;
+    };
+
+    while (pixel < end) {
+        auto symbol = TRY(group[0].read_symbol(bit_stream));
+        dbgln_if(WEBP_DEBUG, "  pixel sym {}", symbol);
+        if (symbol >= 256u + 24u + color_cache_size)
+            return context.error("WebPImageDecoderPlugin: Symbol out of bounds");
+
+        // "1. if S < 256"
+        if (symbol < 256u) {
+            // "a. Use S as the green component."
+            u8 g = symbol;
+
+            // "b. Read red from the bitstream using prefix code #2."
+            u8 r = TRY(group[1].read_symbol(bit_stream));
+
+            // "c. Read blue from the bitstream using prefix code #3."
+            u8 b = TRY(group[2].read_symbol(bit_stream));
+
+            // "d. Read alpha from the bitstream using prefix code #4."
+            u8 a = TRY(group[3].read_symbol(bit_stream));
+
+            emit_pixel(Color(r, g, b, a).value());
+        }
+        // "2. if S >= 256 && S < 256 + 24"
+        else if (symbol < 256u + 24u) {
+            auto prefix_value = [&bit_stream](u8 prefix_code) -> ErrorOr<u32> {
+                // https://developers.google.com/speed/webp/docs/webp_lossless_bitstream_specification#522_lz77_backward_reference
+                if (prefix_code < 4) {
+                    return prefix_code + 1;
+                }
```

Comment:
You can drop the brackets here as well

Reply:
Also copied from the spec. But sure, done.